parsl/usage_tracking/usage.py

```diff
@@ -110,28 +110,46 @@ def __init__(self, dfk, port=50077,
         self.python_version = "{}.{}.{}".format(sys.version_info.major,
                                                 sys.version_info.minor,
                                                 sys.version_info.micro)
-        self.tracking_enabled = self.check_tracking_enabled()
-        logger.debug("Tracking status: {}".format(self.tracking_enabled))
-
-    def check_tracking_enabled(self):
-        """Check if tracking is enabled.
-
-        Tracking will be enabled unless either of these is true:
-
-            1. dfk.config.usage_tracking is set to False
-            2. Environment variable PARSL_TRACKING is set to false (case insensitive)
-
+        self.tracking_level = self.check_tracking_level()
+        self.start_time = None
+        logger.debug("Tracking level: {}".format(self.tracking_level))
+
+    def check_tracking_level(self) -> int:
+        """Check if tracking is enabled and return level.
+
+        Checks the following in order:
+            1. PARSL_TRACKING environment variable
+                - Possible values:
+                    ["true", "false", "True", "False", "0", "1", "2", "3"]
+                - Other values are treated as Level 0 (disabled)
+
+            2. usage_tracking in Config
+                - Possible values: [True, False, 0, 1, 2, 3]
+
+            Config.usage_tracking overrides PARSL_TRACKING if both are set.
```

Comment:
I think this is not entirely true: I set PARSL_TRACKING=3, and Config(usage_tracking=0) and I see two UDP packets being sent.

Reply:
Since I chose to override env variable if usage_tracking is set in config, we cannot differentiate setting```usge_tracking = 0```in the config and usage tracking being disabled by default. So in this specific case I had the env variable override the config. 

I can fix the unnecessary confusing by having PARSL_TRACKING env variable override what's set in the config if usage_tracking is set in both places. What do you think?

Reply:
another policy with that extra "not set" info available would be "parsl will choose the lowest level that is specified by any of the parameters, and 0 if none of them are set"

Reply:
That symmetric policy appeals to me because it's symmetric... There's no arbitrary priority of one setting over the other. And it uses the lowest value that the user has chosen, so there is no "I set a low level but high level info got reported" behaviour.


I think you could compute it something like take the min of (+infinity, usage_tracking if set, PARSL_TRACKING if set) and it you end up with +infinity, that means default to 0

Reply:
Made changes to set usage_tracking parameter default to `None` and choose the lowest level that is specified by any of the parameters, and 0 if none of them are set